File: 6KYU/Equal Sides Of An Array/solution.cs
using System.Linq;
public class Kata
{
  public static int FindEvenIndex(int[] arr)
  {
    for(int n = 0; n < arr.Length; n++)
	  {
		   var left = arr.Where((_, i) => i < n).Sum();
		   var right = arr.Where((_, i) => i > n).Sum();
		   if(left == right)
			    return n;
	  }
	  return -1;
  }
}

File: 6KYU/Equal Sides Of An Array/solution.go
package kata

func Sum(arr []int) int {
  total := 0
  l := len(arr)
  for i:=0; i < l; i++ {
    total += arr[i]
  } 
  return total
}

func FindEvenIndex(arr []int) int {
  l := len(arr)
  
  for i:=0; i < l; i++ {
     if i == 0 && Sum(arr[1:]) == 0 {
         return 0
     }
     if Sum(arr[:i]) == Sum(arr[i+1:]) {
       return i
     }
  } 
  
  return -1
}
